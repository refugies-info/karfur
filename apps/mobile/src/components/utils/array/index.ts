export { default as isLastChild } from "./isLastChild";
