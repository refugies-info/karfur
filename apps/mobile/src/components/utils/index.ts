export * from "./array";
export { isDarkColor } from "./isDarkColor";
