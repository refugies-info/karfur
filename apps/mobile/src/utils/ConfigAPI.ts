import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

export const dbURL = Config.dbUrl;
export const headers = {
  "Content-Type": "application/json",
  "site-secret": Config.siteSecret || "",
};
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers,
});