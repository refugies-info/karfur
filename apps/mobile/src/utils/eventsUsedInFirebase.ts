export enum FirebaseEvent {
  REACH_END_CONTENT = "REACH_END_CONTENT",
  VALIDATE_LANGUAGE = "VALIDATE_LANGUAGE",
  VALIDATE_LOCATION = "VALIDATE_LOCATION",
  VALIDATE_AGE = "VALIDATE_AGE",
  VALIDATE_FRENCH_LEVEL = "VALIDATE_FRENCH_LEVEL",
  PROFILE_COMPLETED = "PROFILE_COMPLETED",
  PROFILE_NOT_COMPLETED = "PROFILE_NOT_COMPLETED",
  CLIC_THEME = "CLIC_THEME",
  CLIC_NEED = "CLIC_NEED",
  CLIC_CONTENT = "CLIC_CONTENT",
  CLIC_LIST = "CLIC_LIST",
  CLIC_CAROUSEL = "CLIC_CAROUSEL",
  CLIC_ACCORDION_ENGAGEMENT = "CLIC_ACCORDION_ENGAGEMENT",
  CLIC_ACCORDION = "CLIC_ACCORDION",
  CLIC_SEE_WEBSITE = "CLIC_SEE_WEBSITE",
  CLIC_SEE_MAP = "CLIC_SEE_MAP",
  SHARE_CONTENT = "SHARE_CONTENT",
  ADD_FAVORITE = "ADD_FAVORITE",
  SWITCH_LANGUAGE = "SWITCH_LANGUAGE",
  LONG_PRESS_CHANGE_LANGUAGE = "LONG_PRESS_CHANGE_LANGUAGE",
  START_VOICEOVER = "START_VOICEOVER",
  OPEN_NOTIFICATION = "OPEN_NOTIFICATION",
  CLIC_NOTIFICATION_ICON = "CLIC_NOTIFICATION_ICON",
  TOGGLE_NOTIFICATION_TYPE = "TOGGLE_NOTIFICATION_TYPE",
  TOGGLE_NOTIFICATION_THEME = "TOGGLE_NOTIFICATION_THEME",
}
