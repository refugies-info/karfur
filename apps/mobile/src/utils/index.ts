export { default as invariant } from "./invariant";
export * from "./types";
