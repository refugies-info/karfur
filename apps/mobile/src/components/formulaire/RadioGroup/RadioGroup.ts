import { View } from "react-native";
import { withProps } from "../../../utils";

const RadioGroup = withProps({ accessibilityRole: "radiogroup" })(View);
export default RadioGroup;
