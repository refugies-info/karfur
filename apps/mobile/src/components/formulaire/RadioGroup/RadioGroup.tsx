import React from "react";
import { View } from "react-native";
import { PropsOf } from "~/utils";

const RadioGroup: React.FC<React.PropsWithChildren<PropsOf<typeof View>>> = ({ children, ...other }) => {
  return (
    <View accessibilityRole="radiogroup" {...other}>
      {children}
    </View>
  );
};

export default RadioGroup;
