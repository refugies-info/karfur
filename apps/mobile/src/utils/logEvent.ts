/* eslint-disable no-console */
import analytics from "@react-native-firebase/analytics";
import Config from "~/libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (eventName: FirebaseEvent, data: Record<string, any>) => {
  const { envName, debugModeFirebase } = Config;
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase) {
      console.log("Environment is dev and debug mode is activated for firebase");
    } else {
      console.log("Environment is dev and debug mode is NOT activated for firebase");
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await analytics().logEvent(eventName, { ...data, envName });
  }
};
