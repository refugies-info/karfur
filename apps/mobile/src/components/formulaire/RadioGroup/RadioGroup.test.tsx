import { Text } from "react-native";
import { render } from "../../utils/tests";
import RadioGroup from "./RadioGroup";

describe("RadioGroup snapshot test suite", () => {
  it("should render without bug", async () => {
    const test = await render(
      <RadioGroup>
        <Text>Test</Text>
      </RadioGroup>,
    );
    expect(test).toMatchSnapshot();
  });
});
