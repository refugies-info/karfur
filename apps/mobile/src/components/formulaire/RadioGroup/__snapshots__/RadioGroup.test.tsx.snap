// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup snapshot test suite should render without bug 1`] = `
<View
  accessibilityRole="radiogroup"
>
  <Text>
    Test
  </Text>
</View>
`;
